stdlib/3.4/asyncio/tasks.pyi

```diff
@@ -19,10 +24,24 @@ def as_completed(fs: Sequence[_FutureT[_T]], *, loop: AbstractEventLoop = ...,
 def ensure_future(coro_or_future: _FutureT[_T],
                   *, loop: AbstractEventLoop = ...) -> Future[_T]: ...
 async = ensure_future
-# TODO: gather() should use variadic type vars instead of _TAny.
-_TAny = Any
-def gather(*coros_or_futures: _FutureT[_TAny],
-           loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[List[_TAny]]: ...
+@overload
+def gather(coro_or_future1: _FutureT[_T1],
+           loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[Tuple[_T1]]: ...
```

Comment:
Somebody could be doing
```
lst = await asyncio.gather(some_coro(), some_other_coro())
lst.append(some_value)
```
I don't think I would write code like that, but somebody might. We have to decide whether the advantage of getting more precise types in the common case outweighs the risk of false positive type errors on code like this.